plugins/system/schema/schema.php

```diff
@@ -0,0 +1,166 @@
+
+ * @phpcs:disable PSR1.Classes.ClassDeclaration.MissingNamespace
+ */
+
+use Joomla\CMS\Application\CMSApplicationInterface;
+use Joomla\CMS\Form\FormHelper;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\Event\EventInterface;
+use Joomla\Event\SubscriberInterface;
+use Joomla\CMS\Event\AbstractEvent;
+use Joomla\CMS\Plugin\PluginHelper;
+
+/**
+ * Schemaorg System Plugin
+ *
+ * @since  4.0.0
+ */
+class PlgSystemSchema extends CMSPlugin implements SubscriberInterface
+{
+    /**
+     * @var    \Joomla\Database\DatabaseDriver
+     *
+     */
+    protected $db;
+
+    /**
+     * Load the language file on instantiation.
+     *
+     * @var    boolean
+     * @since  4.0.0
+     */
+    protected $autoloadLanguage = true;
+
+    /**
+     * Loads the CMS Application for direct access
+     *
+     * @var   CMSApplicationInterface
+     * @since 4.0.0
+     */
+    protected $app;
+
+    /**
+     * The name of the supported name to check against
+     *
+     * @var   string
+     * @since 4.0.0
+     */
+    protected $supportFunctionality = 'core.state';
+
+    /**
+     * Returns an array of events this subscriber will listen to.
+     *
+     * @return  array
+     *
+     * @since   4.0.0
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'onBeforeCompileHead'                  => 'onBeforeCompileHead',
+            'onContentPrepareData'            => 'onContentPrepareData',
+            'onContentPrepareForm'            => 'onContentPrepareForm',
+            'onContentBeforeSave'            => 'onContentBeforeSave',
+        ];
+    }
+
+    /**
+     * Runs on content preparation
+     *
+     * @param   EventInterface  $event  The event
+     *
+     * @since   4.0.0
+     *
+     */
+    public function onContentPrepareData(EventInterface $event)
+    {
+        $context = $event->getArgument('0');
+        $data = $event->getArgument('1');
+
+        // Check if we are manipulating a valid form.
+        if (!in_array($context, ['com_content.article'])) {
+            return true;
+        }
+
+        return true;
+    }
+
+    /**
+     * The form event.
+     *
+     * @param   EventInterface  $event  The event
+     *
+     * @since   4.0.0
+     */
+    public function onContentPrepareForm(EventInterface $event)
+    {
+        $form = $event->getArgument('0');
+
+        // Check if we are manipulating a valid form
+        $context = $form->getName();
+
+        if (!in_array($context, ['com_content.article'])) {
+            return true;
+        }
+
+        //Load the form fields
+        FormHelper::addFormPath(__DIR__ . '/forms');
+        $form->loadFile('schema');
+
+        $event   = AbstractEvent::create(
+            'onSchemaPrepareForm',
+            [
+                'subject' => $form,
+            ]
+        );
+
+        PluginHelper::importPlugin('schemaorg');
+        $this->app->getDispatcher()->dispatch('onSchemaPrepareForm', $event);
```

Comment:
This should be in the new event format.
Two example:
- https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/Workflow/WorkflowServiceTrait.php#L93-L104
- https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/Workflow/Workflow.php#L361-L396